Source/DafnyDriver/CompilerDriver.cs

```diff
@@ -24,365 +23,81 @@
 using Microsoft.Boogie;
 using Bpl = Microsoft.Boogie;
 using System.Diagnostics;
-using DafnyCore.Verifier;
-using Microsoft.Dafny;
-using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.Dafny.LanguageServer.CounterExampleGeneration;
 using Microsoft.Dafny.Plugins;
 
 namespace Microsoft.Dafny {
 
-  public class DafnyDriver : IDisposable {
-    public DafnyOptions Options { get; }
-
+  /// <summary>
+  /// Calls into different phases of Dafny's compilation pipeline,
+  /// such as parsing, resolution, verification and code generation
+  /// 
+  /// Will be replaced by CompilationManager
+  /// </summary>
+  public class CompilerDriver : IDisposable {
```

Comment:
Slightly uncomfortable with the new name, because it feels a bit subtle that this runs verification as well as the "compiler". It's especially weird that `VerifyCommand` calls `RunCompiler`, with options such that no "compilation" happens. :)

Perhaps `DafnyPipelineDriver` instead?

Reply:
Verification requires running the compiler 🤷‍♂️

Reply:
Fair enough, I'll admit I'm biased by the fact that we've (incorrectly) used the word "compiler" for just the code generation phase of the overall compilation pipeline. :) Gradually fixing that will help.

> So translating Dafny to SMT is compilation

More importantly I agree with your point because the verification is just a heavyweight static analysis early on in the overall compilation pipeline (which happens to work by compiling to a different language), and there are good use cases for ONLY running that static analysis by itself.